frontend/components/bridge/utils.ts

```diff
@@ -41,78 +45,95 @@ const getFromToken = (
  * Helper to get bridge refill requirements parameters
  * based on current refill requirements
  */
-export const getBridgeRequirementsParams = ({
-  toMiddlewareChain,
-  fromAddress,
-  toAddress,
-  refillRequirements,
-}: {
-  toMiddlewareChain: MiddlewareChain;
-  fromAddress: Address;
-  toAddress: Address;
-  refillRequirements: AddressBalanceRecord;
-}) => {
-  // TODO: make dynamic, get from token config
-  const fromChainConfig = ETHEREUM_TOKEN_CONFIG;
-  const toChainConfig = TOKEN_CONFIG[asEvmChainId(toMiddlewareChain)];
-
-  const bridgeRequests: BridgeRefillRequirementsRequest['bridge_requests'] = [];
-
-  // Populate bridge requests from refill Requirements
-  for (const [walletAddress, tokensWithRequirements] of Object.entries(
-    refillRequirements,
-  )) {
-    // Only calculate the refill requirements from master EOA or master safe placeholder
-    const isRecipientAddress = areAddressesEqual(walletAddress, toAddress);
-    if (!(isRecipientAddress || walletAddress === 'master_safe')) {
-      continue;
-    }
-
-    for (const [tokenAddress, amount] of Object.entries(
-      tokensWithRequirements,
-    )) {
-      if (!isAddress(tokenAddress)) continue;
-
-      const fromToken = getFromToken(
-        tokenAddress,
-        fromChainConfig,
-        toChainConfig,
-      );
-
-      const fromChain = MiddlewareChain.ETHEREUM;
-      const toChain = toMiddlewareChain;
-      const toToken = tokenAddress as Address;
-
-      const existingRequest = bridgeRequests.find(
-        (req) =>
-          req.from.chain === fromChain &&
-          req.to.chain === toChain &&
-          areAddressesEqual(req.from.address, fromAddress) &&
-          areAddressesEqual(req.to.address, toAddress) &&
-          areAddressesEqual(req.from.token, fromToken) &&
-          areAddressesEqual(req.to.token, toToken),
-      );
-
-      if (existingRequest) {
-        existingRequest.to.amount = (
-          BigInt(existingRequest.to.amount) + BigInt(amount)
-        ).toString();
-      } else {
-        bridgeRequests.push({
-          from: { chain: fromChain, address: fromAddress, token: fromToken },
-          to: {
-            chain: toChain,
-            address: toAddress,
-            token: toToken,
-            amount: `${amount}`,
-          },
-        });
+export const useGetBridgeRequirementsParams = () => {
+  const { selectedAgentConfig } = useServices();
+  const { masterEoa } = useMasterWalletContext();
+
+  const toMiddlewareChain = selectedAgentConfig.middlewareHomeChainId;
+  const fromAddress = masterEoa?.address;
+  const toAddress = masterEoa?.address;
+  const { refillRequirements, isBalancesAndFundingRequirementsLoading } =
+    useBalanceAndRefillRequirementsContext();
```

Comment:
we shouldn't alway use master EOA, in future it can be provided by the user. and for funding after onboarding (in low funds flow) we will transfer funds to master safe

as well as for funding with low funds (or manual funding) we shouldn't use the refill requirements, as the values will be provided manually but the user. that's why I made it configurable